File: app/src/main/java/mishka/mingeo/di/component/PumpingActivityComponent.java
package mishka.mingeo.di.component;

import dagger.Component;
import dagger.Subcomponent;
import mishka.mingeo.di.ActivityScope;
import mishka.mingeo.di.module.ActivityModule;
import mishka.mingeo.di.module.PumpingModule;
import mishka.mingeo.view.pumping.PumpingActivity;

@Component(dependencies = {ApplicationComponent.class, ActivityComponent.class},modules = PumpingModule.class)
public interface PumpingActivityComponent {
    void inject(PumpingActivity activity);
}

File: app/src/main/java/mishka/mingeo/di/component/PumpingActivityComponent.kt
package mishka.mingeo.di.component

import dagger.Component
import mishka.mingeo.data.model.Pumping
import mishka.mingeo.di.ActivityScope
import mishka.mingeo.di.module.PumpingModule
import mishka.mingeo.view.pumping.PumpingActivity
import mishka.mingeo.view.pumping.note.NotesFragment

@Component(dependencies = [ActivityComponent::class], modules = [PumpingModule::class])
@ActivityScope
interface PumpingActivityComponent {
    fun inject(activity: PumpingActivity)
    fun inject(fragment: NotesFragment)
//    fun getPumping(): Pumping
}